astropy/utils/metadata.py

```diff
@@ -81,7 +176,18 @@ def merge(left, right, merge_func=concat, metadata_conflicts='warn'):
 
         else:
             try:
-                out[key] = merge_func(left[key], right[key])
+                if merge_func is None:
+                    for left_type, right_type, merge_cls in MERGE_STRATEGY_CLASSES:
+                        if not merge_cls.enabled:
```

Comment:
Context manager was my first thought too, but it's overkill since the only "context" in which this matters is inside the `merge` function, so it might as well be an argument to that (whereas units, etc. change the behavior of lots of different code within a given context in non-obvious ways).